File: Codes/2903-find-indices-with-index-and-value-difference-i.go
/*
 * @Author: LetMeFly
 * @Date: 2024-05-25 12:44:32
 * @LastEditors: LetMeFly
 * @LastEditTime: 2024-05-25 12:55:00
 */
package main

func abs(x int) int {
    if x >= 0 {
        return x;
    }
    return -x;
}

func findIndices(nums []int, indexDifference int, valueDifference int) []int {
    for i := 0; i < len(nums); i++ {
        for j := i; j < len(nums); j++ {
            if j - i >= indexDifference && abs(nums[i] - nums[j]) >= valueDifference {
                return []int {i, j};
            }
        }
    }
    return []int {-1, -1};
}

File: Codes/2903-find-indices-with-index-and-value-difference-i.py
'''
Author: LetMeFly
Date: 2024-05-25 12:39:14
LastEditors: LetMeFly
LastEditTime: 2024-05-25 12:39:23
'''
from typing import List

class Solution:
    def findIndices(self, nums: List[int], indexDifference: int, valueDifference: int) -> List[int]:
        for i in range(len(nums)):
            for j in range(i, len(nums)):
                if j - i >= indexDifference and abs(nums[i] - nums[j]) >= valueDifference:
                    return [i, j]
        return [-1, -1]